pkg/ddl/util/schema_change_notifier.go

```diff
@@ -339,5 +339,39 @@ func (s *SchemaChangeEvent) GetAddPartitioningInfo() (
 	addedPartInfo *model.PartitionInfo,
 ) {
 	intest.Assert(s.tp == model.ActionAlterTablePartitioning)
-	return s.nonPartTableID, s.tableInfo, s.addedPartInfo
+	return s.oldPartTableID, s.tableInfo, s.addedPartInfo
+}
+
+// NewRemovePartitioningEvent creates a schema change event whose type is
+// ActionRemovePartitioning.
+func NewRemovePartitioningEvent(
+	oldPartitionedTableID int64,
+	newSingleTableInfo *model.TableInfo,
```

Comment:
maybe use non-partition table rather than single table in variable name. (or unpartitioned table? I'm not familiar with the terminology)